x/upgrade/types/planinfo/plan_info.go

```diff
@@ -0,0 +1,106 @@
+package planinfo
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	neturl "net/url"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+// PlanInfo is the special structure that the Plan.Info string can be (as json).
+type PlanInfo struct {
```

Comment:
I made this change.

I moved `x/upgrade/types/planinfo/` stuff to `x/upgrades/plan/`, renamed `plan_info.go` to just `info.go`, and `PlanInfo` to just `Info`.

I'm not quite sure how I feel about it, though.
1. `Info` is such a generic word that I feel it'll be easy to get confused about it.
2. The `Plan` struct is still defined by the protos and placed in `x/upgrades/types/upgrade.pb.go` with extra functionality added in `x/upgrades/types/plan.go`. Having this `x/upgrades/plan/` directory without the `Plan` stuff seems confusing. Additionally, it would now feel weird to put the `Plan` stuff into the `x/upgrades/plan/` directory because it'll be referred to as `plan.Plan`.

@robert-zaremba What do you think about leaving it named the way it was, and in the `types/planinfo/` directory until a refactor breaks out all the `types` stuff at the same time?